docs/notes/parameters.md

```diff
@@ -0,0 +1,89 @@
+# Design note: dealing with Module paramaters
+
+## Introduction
+
+In linen we can specify `Module` paramaters either as attributes or method arguments (usually `__call__`).
+Usually the distinction is obvious: Hyperparamaters like the number of filters or the kernel initializer
+are hyper paramaters and call time arguments are the data and some "mode switches" like whether the model
+should be deterministic or use stochastic regularization.
+
+Some cases are however ambigious. Take for example the `Dropout` module.
+We have a number of clear hyper paramaters:
+1. the dropout rate
+2. the axes for which a dropout mask is generated
+
+and some clear call time arguments:
+1. the input that should be masked using dropout
+2. the (optional) rng used to sample the random mask
+
+There is however a final argument that is ambigious. Which is the `deterministic` argument.
+If `deterministic` is true no dropout mask is sampled. This is typically used during model evaluation.
+However, if we pass `eval=True` or `train=False` to a top-level Module. The `deterministic` argument needs
+to be applied everywhere and the boolean argument needs to be passed down to all the layers that might use `Dropout`.
+If `deterministic` is an attribute we might do the following instead:
+
+```python
+from functools import partial
+from flax import linen as nn
+
+class ResidualModel(nn.Module):
+  drop_rate: float
+
+  @nn.compact
+  def __call__(self, x, *, train):
```

Comment:
Does this syntax work? I thought you need a default value for an kwarg that comes after `*` maybe? (I could be wrong)

Reply:
(and does it need to be `*args`?)

Reply:
Oh, this is just, `...`? Like arbitrary other arguments?

Reply:
This is the syntax for a required keyword argument.

Reply:
You learn something every day!